torchsparse/nn/functional/conv.py

```diff
@@ -114,47 +131,46 @@ def conv3d(inputs,
         output_tensor.kernel_maps = inputs.kernel_maps
         output_tensor.check()
     elif not transpose:
-        _stride = str(stride) if isinstance(stride, int) else \
-                str(stride.cpu().numpy().tolist())
-        kernel_map = inputs.kernel_maps.get(
-            'k%s_os%d_s%s_d%d' % (kernel_size, cur_stride, _stride, dilation),
-            None)
-
-        if isinstance(stride, int):
-            do_downsample = stride > 1
-        else:
-            do_downsample = torch.any(stride > 1).item()
+        kernel_map_key = KernelMapConfig(kernel_size, cur_stride, stride,
+                                         dilation)
+        kernel_map = inputs.kernel_maps.get(kernel_map_key, None)
+
+        do_downsample = False
+        for i in range(3):
+            if stride[i] > 1:
+                do_downsample = True
+                break
```

Comment:
Could we maybe do something like `any(x>1 for x in stride)` instead here?

Reply:
Oh yes that's elegant. I'll change it and make a new commit. I did tested some configurations with asymmetric kernel sizes and strides, and the results were as expected.